### PR TITLE
Remove readonly from getopts as it's implicit in the operation.

### PR DESCRIPTION
**Pull request by Martin2112**
Also rename the test cases so they all follow the same pattern and clean them up a bit.

### CODE REVIEW

File: server/interceptor/interceptor.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (tp *trillianProcessor) Before(ctx context.Context, req interface{}) (conte
 
 	if info.getTree {
 		tree, err := trees.GetTree(
-			ctx, tp.parent.admin, info.treeID, trees.NewGetOpts(trees.Admin, info.readonly, info.treeTypes...))
+			ctx, tp.parent.admin, info.treeID, trees.NewGetOpts(trees.Admin, info.treeTypes...))
```
Comment (pavelkalinnikov) [inline server/interceptor/interceptor.go]: Is the `readonly` flag in `info` needed then? Maybe remove it if not.

Comment (Martin2112) [reply server/interceptor/interceptor.go]: Yes, because it's passed to the quota API.

**Comment by Martin2112** [comment]
I think Admin should be read / write and used sparingly.

File: server/interceptor/interceptor.go (hunk 1 above)
Comment (pavelkalinnikov) [reply server/interceptor/interceptor.go]: Ah, ok.

**Pull request merged**